src/core/config/merge.js

```diff
@@ -15,25 +15,38 @@ import deepExtend from "deep-extend"
 
 const merge = (target, ...sources) => {
   let domNode = Symbol.for("domNode")
-  const sourcesWithoutDomNode = []
+  let primaryName = null
```

Comment:
You need to set this to something unique to be able to determine the override at the end of merge function

```suggestion
  let primaryName = Symbol.for('urls.primaryName')
```